### ISSUE TITLE
New-HPOVServerProfileTemplate -ManageBoot switch Parameter was not provided

### ISSUE THREAD

**Issue by Codingviking** [HewlettPackard/POSH-HPOneView#334]
Please fill in as much information as possible to help resolve your issue.

Love the code!!  Thank you!!!!  I think this is an issue but it might be just me...

### Expected Behavior
    Allow template to be created if Manageboot is false and there are network connections defined.  This is needed because Synergy OneView does not allow a choice for booting from SD cards so you must not select manage boot order to get around this issue.

### Actual Behavior
New-HPOVServerProfileTemplate : Bootable Connections 1, 2, 3, 4, 5, 6 were found, however the -ManageBoot switch Parameter was not provided.  Please correct your command syntax and try again.

### Steps to reproduce

#After module is loaded and Connect-HPOVMgmt has run
$TemplateName        = "Syn480cG10_ESXi1"
$TemplateDescription = "Syn480cG10_ESXi1, version 1.0"
$eg                  = Get-HPOVEnclosureGroup -Name "EG_600"
$Baseline            = Get-HPOVBaseline -FileName 'SPP_2017_10_20180324_for_HPE_Synergy_Z7550-96494.iso' -ErrorAction Stop
$con1                = Get-HPOVNetworkSet -Name "NS_A_VMWare" -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 1 -Name 'A_VMWare' -ConnectionType Ethernet 
$con2                = Get-HPOVNetworkSet -Name "NS_B_VMWare" -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 2 -Name 'B_VMWare' -ConnectionType Ethernet 
$con3                = Get-HPOVNetworkSet -Name "NS_A_Prod"   -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 3 -Name 'A_Prod'   -ConnectionType Ethernet 
$con4                = Get-HPOVNetworkSet -Name "NS_B_Prod"   -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 4 -Name 'B_Prod'   -ConnectionType Ethernet 
$con5                = Get-HPOVNetworkSet -Name "NS_A_Backup" -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 5 -Name 'A_Backup' -ConnectionType Ethernet 
$con6                = Get-HPOVNetworkSet -Name "NS_B_Backup" -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 6 -Name 'B_Backup' -ConnectionType Ethernet 
$bl480bios = @(@{id="WorkloadProfile";value="Virtualization-MaxPerformance"},
               @{id="ProcX2Apic";value="Disabled"},
               @{id="MemPatrolScrubbing";value="Disabled"},
               @{id="PowerRegulator";value="StaticHighPerf"},
               @{id="MinProcIdlePower";value="NoCStates"},
               @{id="MinProcIdlePkgState";value="NoState"},
               @{id="EnergyPerfBias";value="MaxPerf"},
               @{id="CollabPowerControl";value="Disabled"},
               @{id="NumaGroupSizeOpt";value="Clustered"},
               @{id="UncoreFreqScaling";value="Maximum"},
               @{id="SubNumaClustering";value="Enabled"},
               @{id="EnergyEfficientTurbo";value="Disabled"},
               @{id="IntelUpiPowerManagement";value="Disabled"} )
             
$params = @{
	Name               = $TemplateName;
	Description        = $TemplateDescription;
    bios                = $true;
    biossettings        = $bl480bios
    ServerHardwareType = $SY480Gen10SHT;
	EnclosureGroup     = $eg;
	Connections        = $con1, $con2, $con3 ,$con4, $con5, $con6;
	Firmware           = $false;
	Baseline           = $Baseline;
	ManageBoot         = $False;
	FirmwareMode       = 'FirmwareAndSoftware'
	BootMode           = "UEFIOptimized";
	PxeBootPolicy      = "IPv4";
	LocalStorage       = $False;
	HideUnusedFlexnics = $True
}
New-HPOVServerProfileTemplate @params | Wait-HPOVTaskComplete

```PowerShell
# 1. Capture verbose output using the Get-HPOVCommandTrace for HPOneView.310 or newer, or append the -Verbose switch to your Cmdlet call.
# 2. Put your code and/or captured output here.
# 3. Remove this comment block before submitting.
```

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion` or `$PSLibraryVersion`):
ApplianceVersion: 4.00.07.334467.02 4.0.1630.2612  C:\Program Files\WindowsPowerShell\Modules\HPOneView.400

HPE OneView Appliance Version (`Get-HPOVVersion -ApplianceVer`):                                       
ApplianceVersion: 4.00.07.334467.02 4.0.1630.2612  C:\Program Files\WindowsPowerShell\Modules\HPOneView.400

Output from `$PSVersionTable` on your Windows Host:

Name                           Value                                                                                                                                                                             
PSVersion                      5.0.10586.117                                                                                                                                                                                    
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0, 5.0.10586.117}                                                                                                                                                              
BuildVersion                   10.0.10586.117                                                                                                                                                                                   
CLRVersion                     4.0.30319.42000                                                                                                                                                                                  
WSManStackVersion              3.0                                                                                                                                                                                              
PSRemotingProtocolVersion      2.3                                                                                                                                                                                              
SerializationVersion           1.1.0.1     

```PowerShell
# Output from $PSVersionTable
```

**Comment by ChrisLynchHPE** [comment]
This is actually a bug with the Cmdlet validation.  I'm working on an update to the library.  For a workaround, you can do the following to each of your Ethernet connections you create:

```PowerShell
$con1.boot = @{priority = "NotBootable"}
```

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 4.00.1672.1789](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.0.1672.1789) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.400/4.00.1672.1789).

Please verify before this issue is closed.

**Comment by Codingviking** [comment]
Thanks Chris!  Unfortunately no joy.  I get this message now when I set Manageboot to $False with no bootorder line:

"New-HPOVServerProfileTemplate : The boot order list contains invalid or duplicate values or does not conform to the boot order specification."

If leave the Manageboot to $false and put in a Bootorder value of "Hard Drive" it creates the template with the Manage boot option checked and Hard drive selected.

**Comment by ChrisLynchHPE** [comment]
I will have to take a look at this further to see why it is still failing for you.  Are you using the same script you provided above?

**Comment by Codingviking** [comment]
I didn't have the $con.boot statements.  I didn't think I needed with the new update.  Should I put them back in?

$TemplateName        = "Syn480cG10_ESXi2"
$TemplateDescription = "Syn480cG10_ESXi2, version 1.0"
$eg                  = Get-HPOVEnclosureGroup -Name "EG_600"
$Baseline            = Get-HPOVBaseline -FileName 'SPP_2017_10_20180324_for_HPE_Synergy_Z7550-96494.iso' -ErrorAction Stop
$con1                = Get-HPOVNetworkSet -Name "NS_A_VMWare" -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 1 -Name 'A_VMWare' -ConnectionType Ethernet 
$con2                = Get-HPOVNetworkSet -Name "NS_B_VMWare" -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 2 -Name 'B_VMWare' -ConnectionType Ethernet 
$con3                = Get-HPOVNetworkSet -Name "NS_A_Prod"   -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 3 -Name 'A_Prod'   -ConnectionType Ethernet 
$con4                = Get-HPOVNetworkSet -Name "NS_B_Prod"   -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 4 -Name 'B_Prod'   -ConnectionType Ethernet 
$con5                = Get-HPOVNetworkSet -Name "NS_A_Backup" -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 5 -Name 'A_Backup' -ConnectionType Ethernet 
$con6                = Get-HPOVNetworkSet -Name "NS_B_Backup" -ErrorAction Stop | New-HPOVProfileConnection -ConnectionId 6 -Name 'B_Backup' -ConnectionType Ethernet 


$bl480bios = @(@{id="WorkloadProfile";value="Virtualization-MaxPerformance"},
               @{id="ProcX2Apic";value="Disabled"},
               @{id="MemPatrolScrubbing";value="Disabled"},
               @{id="PowerRegulator";value="StaticHighPerf"},
               @{id="MinProcIdlePower";value="NoCStates"},
               @{id="MinProcIdlePkgState";value="NoState"},
               @{id="EnergyPerfBias";value="MaxPerf"},
               @{id="CollabPowerControl";value="Disabled"},
               @{id="NumaGroupSizeOpt";value="Clustered"},
               @{id="UncoreFreqScaling";value="Maximum"},
               @{id="SubNumaClustering";value="Enabled"},
               @{id="EnergyEfficientTurbo";value="Disabled"},
               @{id="IntelUpiPowerManagement";value="Disabled"} )



$params = @{
	Name               = $TemplateName;
	Description        = $TemplateDescription;
    bios                = $true;
    biossettings        = $bl480bios
    ServerHardwareType = $SY480Gen10SHT;
	EnclosureGroup     = $eg;
	Connections        = $con1, $con2, $con3 ,$con4, $con5, $con6;
	Firmware           = $false;
	Baseline           = $Baseline;
	ManageBoot         = $False;
	FirmwareMode       = 'FirmwareAndSoftware'
	BootMode           = "UEFIOptimized";
	PxeBootPolicy      = "IPv4";
	LocalStorage       = $False;
	HideUnusedFlexnics = $True
}

# Create Server Profile Template
New-HPOVServerProfileTemplate @params | Wait-HPOVTaskComplete

**Comment by ChrisLynchHPE** [comment]
No, you do not need to manually set `$con.boot` as the `New-HPOVServerProfileConnection` helper Cmdlet is supposed to properly set that value.  I'll take your script and try to reproduce.

**Comment by antonindostalek** [comment]
Hello, thank you for you very usefull cmdlets.
I faced the same issue as @Codingviking.

After check function New-HPOVServerProfileTemplate it seems that issue is on line 82343, where you fill manageMode    = $true; 
switch ($BootMode), match parameter { "UEFI","UEFIOptimized" -match $_ } 

I think you shoud read param ManageBoot and fill the user value after switch, instead of hard use $false or $true.

BR
Antonin

**Comment by Codingviking** [comment]
Thanks for the feedback!

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 4.10.1821.1567](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.10.1821.1567) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.410/4.10.1821.1567), and also [Release 4.0.1821.1629](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.0.1821.1629) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.400/4.0.1821.1629).

Please verify before this issue is closed.
